packages/trip-form/src/MetroModeSelector/utils.ts

```diff
@@ -0,0 +1,311 @@
+  modeSettings: ModeSetting[]
+): ModeSettingValues {
+  return modeSettings?.reduce((prev, cur) => {
+    prev[cur.key] = cur.default;
+    if (cur.type === "SLIDER" && cur.inverseKey && cur.default) {
+      prev[cur.inverseKey] = cur.high - cur.default + cur.low;
+    }
+    return prev;
+  }, {});
+}
+
+/**
+ * This function is used to apply the ModeSettings to the ModeButtons by checking
+ * each setting against all the transport modes in the button. It also handles the special
+ * case of a "TRANSIT" mode setting, which can apply to all of the different possible TRANSIT_MODES.
+ * @param setting Mode setting to check
+ * @param mode TransportMode to check against
+ * @returns Whether this mode setting applies to this TransportMode
+ */
+export function checkIfModeSettingApplies(
+  setting: ModeSetting,
+  mode: TransportMode
+): boolean {
+  if (setting.applicableMode === "TRANSIT") {
+    return TRANSIT_SUBMODES_AND_TRANSIT.includes(mode.mode);
+  }
+  return setting.applicableMode === mode.mode;
+}
+
+/**
+ * Higher order function that can be used in `map` to add mode settings to mode button definitions.
+ * @param settings Mode settings to be added to button
+ * @returns Function that accepts a mode button definition, returning mode button def with populated settings
+ */
+export const addSettingsToButton = (settings: ModeSetting[]) => (
+  button: ModeButtonDefinition
+): ModeButtonDefinition => {
+  const settingsForThisCombination = Array.from(
+    new Set(
+      button.modes?.reduce<ModeSetting[]>((prev, mode) => {
+        return [
+          ...prev,
+          ...settings.filter(def => checkIfModeSettingApplies(def, mode))
+        ];
+      }, [])
+    )
+  );
+
+  return {
+    ...button,
+    modeSettings: settingsForThisCombination
+  };
+};
+
+/**
+ * State storage hook that can either store the state in React useState hook or in the
+ * the URL Query Params using useQueryParam library. The reason this exists is so that the mode
+ * selector can function in the storybook where modifying the URL doesn't work. It also
+ * helps handle the initial value for when the URL doesn't have the associated query param.
+ * @param name Name of parameter
+ * @param stateType Type of object to be stored
+ * @param storeInQueryParam Store this object in query params or in React state hook
+ * @param defaultState The default state/initial state
+ * @returns Getter and setter just like useState
+ */
+function useStateStorage<Type>(
+  name: string,
+  stateType: QueryParamConfig<Type>,
+  storeInQueryParam: boolean,
+  defaultState?: Type
+): [Type, (newValue: Type) => void] {
+  const [qpState, setQpState] = useQueryParam<Type>(name, stateType);
+  const [reactState, setReactState] = useState<Type>(defaultState);
+
+  if (!storeInQueryParam) {
+    return [reactState, setReactState];
+  }
+
+  if (qpState === undefined) {
+    return [defaultState, setQpState];
+  }
+  return [qpState, setQpState];
```

Comment:
Consider this for a fix PR: As written, if a top-level mode is not activated, the query parameters for that mode still appear in the URL bar. Consider removing query parameters from the URL for unactivated modes.